tests/lib/rules/quotes.js

```diff
@@ -160,30 +160,35 @@ ruleTester.run("quotes", rule, {
             code: "var foo = 'bar';",
             output: "var foo = `bar`;",
             options: ["backtick"],
+            parserOptions: { ecmaVersion: 2015 },
```

Comment:
what if the parserOptions is setting: `{ ecmaVersion: 5 }`. (I would suggest to report a linting error, but do not apply the fix.)

Reply:
It's the thing that this PR fixed -- the test case will fail as "A fatal parsing error occurred in autofix: Parsing error: unexpected token `"

Reply:
to clarify, the parsing error will be reported in rule tester, but not to end users (since eslint don't verify the "fixed" code if I'm not misunderstanding)

Reply:
ESLint repeats autofix until all fixable problems are processed. In this case, ESLint encounters a syntax error while the loop, the syntax error will be reported to users.

Reply:
well, I see. thanks!